core/app/models/workarea/order.rb

```diff
@@ -253,18 +252,23 @@ def add_item(attributes)
       save
     end
 
-    # Updates an items attributes
+    # Update an item's attributes. Increases quantity by the provided
+    # value if a `:quantity` param is passed in.
     #
     # @param [String] id
-    # @param [Hash] attributes new item attributes
+    # @param [Hash] attributes - new item attributes
     #
     # @return [Boolean]
     #   whether the item was successfully updated
     def update_item(id, attributes)
-      existing_item = items.find_existing(attributes[:sku], attributes[:customizations])
+      existing_item = items.find_existing(attributes[:sku], attributes)
+
       if existing_item.present? && existing_item.id.to_s != id.to_s
         item = items.find(id)
-        existing_item.update_attributes(quantity: existing_item.quantity + (attributes[:quantity] || item.quantity))
+        attributes[:quantity] = existing_item.quantity + (
```

Comment:
Changing `attributes` here is unnecessary mutation of the hash that's been passed in, let's leave this as it was.

Reply:
yeah originally i had `dup`-ed it but then i realized we probably don't want to update attrs in `update_item`.